File: src/types/provider.ts
export interface LlmProvider {
  id: string;
  name: string;
  baseUrl: string;
  isActive: boolean;
  createdAt: string;
  updatedAt: string;
}

export interface LlmModel {
  id: string;
  providerId: string;
  modelId: string;
  name: string;
  isDefault: boolean;
  createdAt: string;
}

export interface CreateProviderPayload {
  name: string;
  baseUrl: string;
  apiKey: string;
}

export interface SetDefaultModelPayload {
  providerId: string;
  modelId: string;
  modelName: string;
}

File: src-tauri/src/models/provider.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct LlmProvider {
    pub id: String,
    pub name: String,
    pub base_url: String,
    pub is_active: bool,
    pub created_at: String,
    pub updated_at: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct LlmModel {
    pub id: String,
    pub provider_id: String,
    pub model_id: String,
    pub name: String,
    pub is_default: bool,
    pub created_at: String,
}

#[derive(Debug, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct CreateProviderPayload {
    pub name: String,
    pub base_url: String,
    pub api_key: String,
}

#[derive(Debug, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct SetDefaultModelPayload {
    pub provider_id: String,
    pub model_id: String,
    pub model_name: String,
}
